data_steward/cdr_cleaner/cleaning_rules/base_cleaning_rule.py

```diff
@@ -175,6 +170,7 @@ def __init__(self,
         self._sandbox_dataset_id = sandbox_dataset_id
         self._issue_urls = issue_urls if issue_urls else []
         self._depends_on_classes = depends_on if depends_on else []
+        self._affected_tables = affected_tables if affected_tables else []
```

Comment:
I think I would set this to something like:
```
self.set_affected_tables(affected_tables)
```
Then the value can be validated by the argument validation methods.
```
def set_affected_tables(self, affected_tables):
    self.__affected_tables = affected_tables if affected_tables else []
```
Then, classes that need to can override this with a more appropriate implementation.  For most classes, this will suffice.

Reply:
Removing the above class variable is giving making it unable to implement the suggested setter.